sda/internal/storage/storage_test.go

```diff
@@ -323,11 +323,17 @@ func (suite *StorageTestSuite) TestS3Backend() {
 	assert.NoError(suite.T(), err, "s3 NewFileReader failed when it should work")
 	assert.NotNil(suite.T(), reader, "Reader that should be usable is not, bailing out")
 
-	size, err := s3back.GetFileSize("s3Creatable")
+	size, err := s3back.GetFileSize("s3Creatable", false)
 	assert.Nil(suite.T(), err, "s3 GetFileSize failed when it should work")
 	assert.NotNil(suite.T(), size, "Got a nil size for s3")
 	assert.Equal(suite.T(), int64(len(writeData)), size, "Got an incorrect file size")
 
+	// make sure expectDelay=true works
+	size, err = s3back.GetFileSize("s3Creatable", true)
+	assert.Nil(suite.T(), err, "s3 GetFileSize with expected delay failed when it should work")
+	assert.NotNil(suite.T(), size, "Got a nil size for s3")
+	assert.Equal(suite.T(), int64(len(writeData)), size, "Got an incorrect file size")
+
```

Comment:
Shouldn't this use a freshly uploaded file to realy see if it works as expected?

Reply:
Added in 496df2d (and then rebased on main).